### PR TITLE
SQL_DDL_HW

### PR DESCRIPTION
**Pull request by Volod980**

### CODE REVIEW

**Review by albusel — CHANGES_REQUESTED** [review]
Hi Volodymyr,

Current state wouldn't be allowed to deploy on production for the following reasons.

Script isn't rerunnable.

No 'IF NOT EXISTS' used for tables creation

Primary keys are hardcoded

No checks if a value already exists 

Hardcode is used everywhere, instead of "INSERT INTO ... SELECT" "INSERT INTO ... VALUES" is used

No comments at all

It would be great to fix the point above, thanks

**Comment by Volod980** [comment]
Hello, I  put updated file "SQL_DDL_Volodymyr_Kotelnytskyi_HW_UPD". Thank you so much!

File: SQL_DDL_Volodymyr_Kotelnytskyi_HW_UPD.sql [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+-- create table "station"
+create table if not exists  subway.station (station_id serial primary key unique,
+						     name varchar(100) not null ,
+						     is_transfer boolean not null default false,
+						     line_id int not null references subway.line(line_id));
+						   
+-- add information to table "station"							    
+insert into subway.station ( name, is_transfer, line_id)
+select 'Vokzalna' as name, true as is_transfer, 1 as line_id
```
Comment (albusel) [inline SQL_DDL_Volodymyr_Kotelnytskyi_HW_UPD.sql]: Hardcoded line_id
Select statement should be used to find line_id

File: SQL_DDL_Volodymyr_Kotelnytskyi_HW_UPD.sql [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+
+-- add information to table "employee"							 
+insert into subway.employee (first_name, last_name, date_of_birth, position_id, station_id, start_date, end_date) 							 
+select 'Volodymyr' as first_name, 'Kotelnytskyi' as  last_name,date('2003-11-26') as date_of_birth, 1 as position_id, 1 as station_id,date('2020-01-01') as start_date, date('2024-09-23') as end_date							 
+where not exists (select * from subway.employee s
+                  where s.first_name = first_name
+                  and s.last_name =  last_name
+                  and s.date_of_birth = date_of_birth
+                  and s.position_id = position_id
```
Comment (albusel) [inline SQL_DDL_Volodymyr_Kotelnytskyi_HW_UPD.sql]: Hardcode

**Review by albusel — CHANGES_REQUESTED** [review]
Hardcode is still in place, let's get rid of it!

**Comment by albusel** [comment]
and please add LOWER/UPPER function for strings

**Comment by Volod980** [comment]
Hello, thank you very much for comments. I changed file SQL_DDL_Volodymyr_Kotelnytskyi_HW_UPD

**Review by albusel — APPROVED** [review]
Approved, you can merge

**Pull request merged**